File: Clases/02-POO/Auto-Moto.py
class Vehiculo:
    def __init__(self, marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis):
        self.marca = marca
        self.patente = patente
        self.modelo = modelo
        self.ruedas = ruedas
        self.combustible = combustible
        self.kilometraje = kilometraje
        self.color = color
        self.chasis = chasis

    def avanza(self):
        print("El vehículo avanza")

    def gira(self):
        print("El vehículo gira")

    def detiene(self):
        print("El vehículo se detiene")

    def enciende(self):
        print("El vehículo se enciende")

    def apaga(self):
        print("El vehículo se apaga")


class Auto(Vehiculo):
    def __init__(self, marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis, puertas, caja_cambio):
        super().__init__(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis)
        self.puertas = puertas
        self.caja_cambio = caja_cambio

    def retrocede(self):
        print("El auto retrocede")


class Moto(Vehiculo):
    def __init__(self, marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis):
        super().__init__(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis)

    def enciende(self):
        print("La moto se enciende")


class Camion(Vehiculo):
    def __init__(self, marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis):
        super().__init__(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis)

File: Clases/02-POO/Auto-Moto.js
// Clase Vehiculo
class Vehiculo {
    constructor(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis) {
        this.marca = marca;
        this.patente = patente;
        this.modelo = modelo;
        this.ruedas = ruedas;
        this.combustible = combustible;
        this.kilometraje = kilometraje;
        this.color = color;
        this.chasis = chasis;
    }

    avanza() {
        console.log(`El vehículo ${this.marca} ${this.modelo} avanza`);
    }

    gira() {
        console.log(`El vehículo ${this.marca} ${this.modelo} gira`);
    }

    detiene() {
        console.log(`El vehículo ${this.marca} ${this.modelo} se detiene`);
    }

    enciende() {
        console.log(`El vehículo ${this.marca} ${this.modelo} se enciende`);
    }

    apaga() {
        console.log(`El vehículo ${this.marca} ${this.modelo} se apaga`);
    }
}

// Clase Auto (hereda de Vehiculo)
class Auto extends Vehiculo {
    constructor(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis, puertas, caja_cambio) {
        super(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis);
        this.puertas = puertas;
        this.caja_cambio = caja_cambio;
    }

    retrocede() {
        console.log(`El auto ${this.marca} ${this.modelo} retrocede`);
    }
}

// Clase Moto (hereda de Vehiculo)
class Moto extends Vehiculo {
    constructor(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis) {
        super(marca, patente, modelo, ruedas, combustible, kilometraje, color, chasis);
    }

    enciende() {
        console.log(`La moto ${this.marca} ${this.modelo} se enciende`);
    }
}

const moto1 = new Moto("Yamaha", "ABC123", "YBR 125", 2, "Gasolina", 0, "Blanca", "YBR 125");
